cranelift/filetests/filetests/egraph/icmp-parameterized.clif

```diff
@@ -206,7 +206,7 @@ block0(v0: i32):
 
 ; function %icmp_eq_smin(i32) -> i8 fast {
 ; block0(v0: i32):
-;     v1 = iconst.i32 0x8000_0000
+;     v1 = iconst.i32 0xffff_ffff_8000_0000
```

Comment:
@jameysharp this is not super clean, the parser could not re-parse this output, I don't know if it's a big deal

Reply:
Yeah, something's a little off about these tests, which appear to be sign-extending 32-bit constants to 64-bit when printing them. I think there's a bug in the writer but I haven't looked closely at it to figure out what that might be.

Reply:
I didn't check but most probably the Display instance for Imm64 is falling back to hex representation for big values. This particular example is a valid negative i32 value, but a big one, so maybe the Display implementation is falling back to unsigned 64-bit hexadecimal as it can't know it's a 32-bit value. We could add range checks in this Display instance and emit the right amount of hex digits; or I could remove my changes in the CLIF writer altogether and figure out another way of fixing the tests it solved.

Reply:
I see now. Before this PR, for constants narrower than 64 bits we were preserving whether the input was written as a negative number or not based on whether the Imm64 was sign-extended or zero-extended. So some of the egraph tests used small negative numbers on input and got negative numbers in the test expectations output; others used large enough positive numbers to have the high bit set, which were printed as positive hex numbers before but get sign-extended by your change to the writer now.

I think we should revert your change to the writer so it doesn't try to sign-extend constants, and then we'll probably just need to update the test expectations to reflect that some negative constants are printed as large positive hex values now.